scheduling-utils/src/pubkeys_ptr.rs

```diff
@@ -0,0 +1,40 @@
+use {
+    agave_scheduler_bindings::SharablePubkeys, rts_alloc::Allocator, solana_pubkey::Pubkey,
+    std::ptr::NonNull,
+};
+
+pub struct PubkeysPtr<'a> {
+    ptr: NonNull<Pubkey>,
+    count: usize,
+    allocator: &'a Allocator,
+}
+
+impl<'a> PubkeysPtr<'a> {
+    /// Constructs the pointer from a [`SharablePubkeys`].
+    ///
+    /// # Safety
+    ///
+    /// - The allocation pointed to by this region must not have previously been freed.
+    /// - Pointer must be exclusive so that calling [`Self::free`] is safe.
+    pub unsafe fn from_sharable_pubkeys(
+        sharable_pubkeys: &SharablePubkeys,
+        allocator: &'a Allocator,
+    ) -> Self {
+        let ptr = allocator.ptr_from_offset(sharable_pubkeys.offset).cast();
+        Self {
+            ptr,
+            count: sharable_pubkeys.num_pubkeys as usize,
+            allocator,
+        }
+    }
+
+    /// Returns the allocation as a slice.
+    pub fn as_slice(&self) -> &[Pubkey] {
+        unsafe { core::slice::from_raw_parts(self.ptr.as_ptr(), self.count) }
```

Comment:
let's add SAFETY comments here, even though it's basically just "constructor has these constraints"

Reply:
c0be153c47abd7b247efd0094d80cea2f9a9cf39